stellargraph/mapper/node_mappers.py

```diff
@@ -448,3 +454,87 @@ def flow_from_dataframe(self, node_targets):
         """
 
         return NodeSequence(self, node_targets.index, node_targets.values)
+
+
+class FullBatchNodeSequence(Sequence):
+    def __init__(self, features, A, targets=None, sample_weight=None):
+        # Check targets is iterable & has the correct length
+        if not is_real_iterable(targets) and targets is not None:
+            raise TypeError("Targets must be None or an iterable or numpy array ")
+
+        self.features = features
+        self.A = A
+        self.targets = targets
+        self.sample_weight = sample_weight
+
+    def __len__(self):
+        return 1
+
+    def __getitem__(self, index):
+        return [self.features, self.A], self.targets, self.sample_weight
+
+
+class FullBatchNodeGenerator:
+    def __init__(self, G, name=None, func_opt=None, **kwargs):
+        if not isinstance(G, StellarGraphBase):
```

Comment:
a doc string is needed here, especially for `func_opt` argument!

Reply:
https://github.com/stellargraph/stellargraph/blob/feature/gcn/stellargraph/mapper/node_mappers.py#L511

Reply:
You also need to throw an exception if `func_opt` is not callable, and write a test for that.